from program import alg1, alg2

graf1 = [[1, 1, 0, 0, 1], 
         [1, 1, 1, 0, 0], 
         [0, 1, 1, 1, 0], 
         [0, 0, 1, 1, 1], 
         [1, 0, 0, 1, 1]]

graf2 = [[1, 1, 1, 0, 0], 
         [1, 1, 1, 0, 0], 
         [1, 1, 1, 0, 0], 
         [0, 0, 0, 1, 1], 
         [0, 0, 0, 1, 1]]

graf3 = [[1, 1, 0, 1, 0, 0, 0], 
         [1, 1, 1, 0, 0, 0, 0], 
         [0, 1, 1, 1, 0, 0, 0], 
         [1, 0, 1, 1, 0, 0, 0], 
         [0, 0, 0, 0, 1, 1, 1], 
         [0, 0, 0, 0, 1, 1, 1], 
         [0, 0, 0, 0, 1, 1, 1]]


graf4 = [[1, 1, 0, 0, 0, 0], 
         [1, 1, 0, 0, 0, 0], 
         [0, 0, 1, 1, 0, 0], 
         [0, 0, 1, 1, 0, 0], 
         [0, 0, 0, 0, 1, 1],
         [0, 0, 0, 0, 1, 1]]


assert alg1(graf1) == 1, 'Ошибка в обработке графа, не верный ответ'
assert alg1(graf2) == 2, 'Ошибка в обработке графа, не верный ответ'
assert alg1(graf3) == 2, 'Ошибка в обработке графа, не верный ответ'
assert alg1(graf4) == 3, 'Ошибка в обработке графа, не верный ответ'

assert alg2(graf1) == 1, 'Ошибка в обработке графа, не верный ответ'
assert alg2(graf2) == 2, 'Ошибка в обработке графа, не верный ответ'
assert alg2(graf3) == 2, 'Ошибка в обработке графа, не верный ответ'
assert alg2(graf4) == 3, 'Ошибка в обработке графа, не верный ответ'

print('Все тесты прошли успешно')